src/main/java/net/imglib2/algorithm/gradient/PartialDerivative.java

```diff
@@ -69,10 +83,82 @@
 		}
 	}
 
+	// parallel version...
+	/**
+	 * Compute the partial derivative of source in a particular dimension.
+	 *
+	 * @param source
+	 *            source image, has to provide valid data in the interval of the
+	 *            gradient image plus a one pixel border in dimension.
+	 * @param gradient
+	 *            output image
+	 * @param dimension
+	 *            along which dimension the partial derivatives are computed
+	 * @param nTasks
+	 *            Number of tasks for gradient computation.
+	 * @param es
+	 *            {@link ExecutorService} providing workers for gradient
+	 *            computation. Service is managed (created, shutdown) by caller.
+	 */
+	public static < T extends NumericType< T > > void gradientCentralDifferenceParallel(
+			final RandomAccessible< T > source,
+			final RandomAccessibleInterval< T > gradient,
+			final int dimension,
+			final int nTasks,
+			final ExecutorService es ) throws InterruptedException, ExecutionException
+	{
+		final int nDim = source.numDimensions();
+		if ( nDim < 2 )
+		{
+			gradientCentralDifference2( source, gradient, dimension );
```

Comment:
No particular reason: I will change that to `gradientCentralDifference` to be consistent with what happens in `HessianMatrix` and use the faster version (according to the comments in `PartialDerivative`.